lib/elastic_sync/schema.ex

```diff
@@ -1,22 +1,29 @@
 defmodule ElasticSync.Schema do
-  defstruct [:index, :type]
+  defstruct [:index, :type, :config]
 
   defmacro __using__(opts) do
     index  = Keyword.get(opts, :index)
     type   = Keyword.get(opts, :type, index)
+    {mod, fun} = Keyword.get(opts, :config, {ElasticSync.Schema, :default_config})
 
     validate_index_name!(index)
+    validate_config!(mod, fun)
```

Comment:
Since it isn't a required option, and since developers will just be entering raw values (not some computed value), I don't think we need to check for nil.